### ISSUE TITLE
pre-train data setting of yoloworld and yoloe

### ISSUE THREAD

**Issue by sea190** [ultralytics/ultralytics#21691]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Recently, I have been using the yoloe and yoloworld models, and I would like to ask what kind of pre-training data the weights of these two models are trained on

**Comment by UltralyticsAssistant** [comment]
👋 Hello @sea190, thanks for opening this issue and for your interest in Ultralytics 🚀! This is an automated response to help you get started quickly. An Ultralytics engineer will also review your question and follow up soon.

We recommend a visit to the Docs for general guidance where you can find many Python and CLI usage examples and where many common questions may already be answered:
- Docs: https://docs.ultralytics.com/
- Python: https://docs.ultralytics.com/usage/python/
- CLI: https://docs.ultralytics.com/usage/cli/

If this is a 🐛 Bug Report, please provide a minimum reproducible example to help us debug it:
- MRE: https://docs.ultralytics.com/help/minimum-reproducible-example/

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our Tips for Best Training Results:
- Training Tips: https://docs.ultralytics.com/guides/model-training-tips/

Join the Ultralytics community where it suits you best for discussion and support:
- Discord (real-time chat) 🎧: https://discord.com/invite/ultralytics
- Discourse (long-form Q&A): https://community.ultralytics.com/
- Subreddit (community threads): https://www.reddit.com/r/Ultralytics/

## Upgrade

Please upgrade to the latest ultralytics package including all requirements in a Python>=3.8 environment with PyTorch>=1.8 to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```
- Requirements: https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including CUDA/CUDNN, Python and PyTorch preinstalled):

- Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Google Cloud Deep Learning VM. See GCP Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/
- Amazon Deep Learning AMI. See AWS Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/
- Docker Image. See Docker Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/ <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all Ultralytics CI tests are currently passing. CI tests verify correct operation of all YOLO Modes and Tasks on macOS, Windows, and Ubuntu every 24 hours and on every commit.

To help us assist you faster on your question about pre-training for YOLOE and YOLOWorld, please share:
- The exact model variants and weights you’re using (links/filenames)
- Your environment details (OS, Python, PyTorch, CUDA versions)
- Any references you’ve checked in the Docs or model cards

Thank you! 🙌

**Comment by Y-T-G** [comment]
You can check the training from scratch section:

https://docs.ultralytics.com/models/yoloe/#launching-training-from-scratch

https://docs.ultralytics.com/models/yolo-world/#launch-training-from-scratch

**Comment by sea190** [comment]
Thanks for reply, The documentation says that the training datasets used for reproduction are Objects365, flickr and gqa, does that mean that the pre-training data for the weights you provided are also these?

**Comment by sea190** [comment]
Thanks for reply, The documentation says that the training datasets used for reproduction are Objects365, flickr and gqa, does that mean that the pre-training data for the weights you provided are also these?

**Comment by glenn-jocher** [comment]
Thanks for the follow-up.  
YOLO‑World: the weights in Ultralytics are the official checkpoints migrated as‑is, so their pre‑training follows the authors’ setup; our Objects365 + GQA + Flickr30k recipe is provided to reproduce from scratch—see the [YOLO‑World docs](https://docs.ultralytics.com/models/yolo-world/).  
YOLOE: the checkpoints we reference are from the YOLOE authors; their training uses segmented Objects365 with grounding datasets and is validated on LVIS—see the dataset table in the [YOLOE docs](https://docs.ultralytics.com/models/yoloe/). For exact curation, please refer to the original papers/repos linked from those pages.

**Comment by caicai0352** [comment]
when performing pre-training with YOLOE, the documentation mentions that the Objects365 dataset provides annotations in COCO format. However, during actual usage, it loads the Objects365.yaml file, which is in the original YOLO format. Are there any additional processing steps required in between? How are these two formats related?

**Comment by glenn-jocher** [comment]
Great question—Objects365 is released in COCO JSON, while `Objects365.yaml` expects YOLO TXT labels; if you’re starting from the official JSON, convert to YOLO format and place outputs under `labels/train` and `labels/val` so the YAML resolves, e.g.:

```python
from ultralytics.data.converter import convert_coco
convert_coco(labels_dir="path/to/objects365/annotations")  # produces YOLO TXT labels
```

For YOLOE pre-training, the extra segmentation annotations are separate JSONs; use the processed ones and pass them via `grounding_data` (you can also include the Objects365 segmentation JSON in the same way). Details are in the YOLOE dataset prep section and the Objects365 dataset guide, and the COCO→YOLO step is shown in the COCO-to-YOLO conversion snippet: [YOLOE: prepare datasets](https://docs.ultralytics.com/models/yoloe/#prepare-datasets), [Objects365 dataset guide](https://docs.ultralytics.com/datasets/detect/objects365/), and [COCO-to-YOLO conversion](https://docs.ultralytics.com/datasets/detect/#port-or-convert-label-formats).
